### ISSUE TITLE
Let NVDA announce all characters which are in font symbol or windings when punctuation is "some"

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7804]
# Steps to reproduce:
1. Open MS word and type following:
1. production process
1.1. planning
• Example
- part one
- part two
2. starting process
2.1. equipment requirements
a. raw materials
b. machines
c. packages

## Expected behavior:
NVDA should read on each line the level of enumeration and the shape of bullets. More exactly, MS word converts each time the numbers and bullets in text font symbols or windings. All characters which are converted by MS word in a bullet or enumeration list should be announced by NVDA even when "punctuation" is set on "some" or "none" in voice settings.

## Actual behavior:
When "Punctuation" is set on "some", "- part one" and "- part two" are not being announced together wit the coresponding bullet point since NVDA does not announce dashes in this punctuation setting. However, if punctuation setting is most, every dash is being announced which is not needed in this case (see atachment). 

# System configuration:
## NVDA version:
All

## NVDA Installed or portable:
both

# Other information:
## Windows version:
all versions

## Name and version of other software in use when reproducing the issue:
Office all Versions since 2007

Does the issue still occur after restarting your PC?
Yes

Have you tried any other versions of NVDA?
Every Version since 2013.2
[production.process.docx](https://github.com/nvaccess/nvda/files/1515582/production.process.docx)

**Comment by feerrenrut** [comment]
Please update the expected and actual steps to include the speech you get and expect from NVDA this helps to complete the example.

**Comment by Adriani90** [comment]
I have changed the steps to reproduce and added some examples in the behaviour section. Hope it is mor understandable now.

**Comment by feerrenrut** [comment]
Yes, I think this behaviour should be changed. Announcement of the list element is currently being stopped by the punctuation level.

This will probably be tricky to fix, essentially punctuation level needs to be set to all before a list item is spoken. Alternatively, some blocks of text (like list items) should not be subject to the punctuation level rules.

**Comment by Adriani90** [comment]
If we linit this to the text font symbols and windings it should not cause any conflicts. What do you think?


Von meinem iPhone gesendet

**Comment by feerrenrut** [comment]
Ideally the name for the "symbol" part of the bullet item is always reported accurately. By this I mean, given "- some-text with-dashes", only the dash that is used as a bullet should be announced. I would suggest that the exact symbol used is not important and NVDA should announce "bullet level 2 some-text with-dashes".

A full example, given:
```
* something that is a level one item
    - something that is a level two item
```
With "lists" unchecked in the document formatting dialog NVDA should announce:
```
level one bullet something that is a level one item
level two bullet something that is a level two item
```

In terms of how we achieve this, I would rather not speculate on the implementation. It will require some investigation by whoever implements it. Hopefully, Word delineates the bullet symbol.

**Comment by Adriani90** [comment]
That’s an interesting idea. However, I thought it would be much easier just to implement a rule for already existing speech rules rather than defining a new speech rule. I mean, what if – is bullet level one and * bullet level 2? How do you sugerate NVDA that – is not level 2 in this case?

 

 

Von: Reef Turner [mailto:notifications@github.com] 
Gesendet: Montag, 11. Dezember 2017 08:28
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Let NVDA announce all characters which are in font symbol or windings when punctuation is "some" or "none" (#7804)

 

Ideally the name for the "symbol" part of the bullet item is always reported accurately. By this I mean, given "- some-text with-dashes", only the dash that is used as a bullet should be announced. I would suggest that the exact symbol used is not important and NVDA should announce "bullet level 2 some-text with-dashes".

A full example, given:

* something that is a level one item
    - something that is a level two item

With "lists" unchecked in the document formatting dialog NVDA should announce:

level one bullet something that is a level one item
level two bullet something that is a level two item

In terms of how we achieve this, I would rather not speculate on the implementation. It will require some investigation by whoever implements it. Hopefully, Word delineates the bullet symbol.

**Comment by feerrenrut** [comment]
Well, that is an implementation detail. The way we get metadata about the contents of a word document likely allows this information to be collected. However, some investigation is required to confirm this. 

One thing to note is that a rules based replacement is likely to be fragile, essentially any character or image can be set as the bullet for a particular level. 

Assuming implementation is not a problem, would the following be preferred, or too verbose (assuming this is something that can not be turned off):
```
level one bullet something that is a level one item
level two bullet something that is a level two item
```

**Comment by Adriani90** [comment]
The only Thing I am struggling with, is following: If I get a document with 5 bullet levels and I have to continue on Level 4, how do I know which sign I must type in order to continue the path? Assume word does not automatically write forward the list Level when pressing enter.

**Comment by feerrenrut** [comment]
In word, the tab key increases the bullet level, and shift tab decreases the bullet level. Shameless plug: our [Microsoft Word with NVDA training material](https://www.nvaccess.org/product/microsoft-word-training-for-nvda) covers this.

**Comment by Adriani90** [comment]
This changes intendation as well which is not always the case when sighted people bullet things. Many people prefer to use different symbols on the same intendation level. But your observation is also very important. I think though the intendation level should be rather reported by centimeters or so. It is better for the user then to visualize how the text looks like. And it should not be reported by default.


I see your point, but I don‘t think it would solve this issue in particular.


Von meinem iPhone gesendet

**Comment by feerrenrut** [comment]
>Many people prefer to use different symbols on the same intendation level

I have to question this, I don't recall seeing a document like that.

The common case is to use tab to increase the bullet level, and yes, also increase the indentation. The symbols used for a bullet level can be changed via a dropdown menu on the bullet button, and any symbol may be selected, for instance 'r'.

You can change the paragraph indentation for different bullet levels, and you could make them all match. Visually this would be quite confusing, and I think quite unlikely. 

In any case, my point is that semantically it's the bullet level that is important rather than the formatting of the bullet.

**Comment by Adriani90** [comment]
I agree with you. In most cases your proposal will actually help the user understand the structure. So in fact, NVDA's alignment reporting in a list of bullet elements should Sound like 

"Level one: * Bullet"
"level 2:         - Bullet"


Without the * and - being reported when punctuation is set on none.
* and - will be reported when punctuation is set on all or according to existing punctuation rules.

This should be implemented for Text Editors, MS Word, Power Point, Outlook and, if possible, pdf and Browsers.

**Comment by dkager** [comment]
Some thoughts:
- I think being able to find out the actual bullet symbol used has its value when you want to follow a certain format in your document, e.g. bullets instead of dashes or circles.
- Bfesides speaking the list level, it really should also be conveyed in braille somehow.

**Comment by Adriani90** [comment]
@dkager I agree. I think bullet shapes could be reported by NVDA regardless of punctuation settings, only when bullet shapes are recognized as list bullets. That means for example, if I use * at the beginning of a line to bullet something, NVDA should report it in any case (i.e. punctuation settin is set on "none"). But if I use * within a text and punctuation setting is set on "none", the star sign which is in the text should not be reported.

**Comment by Adriani90** [comment]
In braille we could show the list level with l1, l2 and so on. Or the tab sign feature from the addon "BrailleExtender" could be implemented in NVDA's core.
